src/main/main.cpp

```diff
@@ -71,15 +71,46 @@ unique_ptr<GPU> create_gpu(MemoryInterface *memory_ptr, CPUInterface *cpu_ptr,
 int main(int argc, char *argv[]) {
 	ios_base::sync_with_stdio(false);
 
+	auto rom_path = string("");
 	if (argc < 2) {
-		cerr << "Please provide the name of a ROM file." << endl;
-		exit(1);
+		cerr << "Please provide the name of a ROM file or press q/Q to exit"
+		     << endl;
+		getline(cin, rom_path);
+		if (rom_path == "q" || rom_path == "Q") {
+			cout << "Exiting TVP" << endl;
+			exit(0);
+		}
+	} else {
+		rom_path = string(argv[1]);
 	}
-	auto rom_path = string(argv[1]);
 
 	auto cartridge = std::make_unique<Cartridge>(rom_path);
+
+	// Reading Meta Data of ROM file and checking if ROM is valid or Not
+	unsigned short int logo_hex_dump[48] = {
+	    0xCE, 0xED, 0x66, 0x66, 0xCC, 0x0D, 0x00, 0x0B, 0x03, 0x73, 0x00, 0x83,
+	    0x00, 0x0C, 0x00, 0x0D, 0x00, 0x08, 0x11, 0x1F, 0x88, 0x89, 0x00, 0x0E,
+	    0xDC, 0xCC, 0x6E, 0xE6, 0xDD, 0xDD, 0xD9, 0x99, 0xBB, 0xBB, 0x67, 0x63,
+	    0x6E, 0x0E, 0xEC, 0xCC, 0xDD, 0xDC, 0x99, 0x9F, 0xBB, 0xB9, 0x33, 0x3E};
+	unsigned int logo_address = 0x0104;
```

Comment:
Use `uint8_t` for 8bit data, and `Address` for 16bit addresses. `Address` is simply an alias for `uint16_t` defined by the memory module.